src/main/java/teammates/ui/webapi/action/CreateFeedbackResponseCommentAction.java

```diff
@@ -0,0 +1,124 @@
+package teammates.ui.webapi.action;
+
+import java.time.Instant;
+import java.util.ArrayList;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.datatransfer.FeedbackParticipantType;
+import teammates.common.datatransfer.attributes.FeedbackResponseAttributes;
+import teammates.common.datatransfer.attributes.FeedbackResponseCommentAttributes;
+import teammates.common.datatransfer.attributes.FeedbackSessionAttributes;
+import teammates.common.datatransfer.attributes.InstructorAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.common.util.Assumption;
+import teammates.common.util.Const;
+import teammates.ui.webapi.output.FeedbackResponseCommentData;
+import teammates.ui.webapi.request.FeedbackResponseCommentSaveRequest;
+
+/**
+ * Creates a new feedback response comment.
+ */
+public class CreateFeedbackResponseCommentAction extends Action {
+
+    @Override
+    protected AuthType getMinAuthLevel() {
+        return AuthType.LOGGED_IN;
+    }
+
+    @Override
+    public void checkSpecificAccessControl() {
+        String feedbackResponseId = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_RESPONSE_ID);
+        FeedbackResponseAttributes response = logic.getFeedbackResponse(feedbackResponseId);
+        Assumption.assertNotNull(response);
+
+        String courseId = response.courseId;
+        String feedbackSessionName = response.feedbackSessionName;
+
+        InstructorAttributes instructor = logic.getInstructorForGoogleId(courseId, userInfo.id);
+        FeedbackSessionAttributes session = logic.getFeedbackSession(feedbackSessionName, courseId);
+
+        gateKeeper.verifyAccessible(instructor, session, response.giverSection,
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS);
+        gateKeeper.verifyAccessible(instructor, session, response.recipientSection,
+                Const.ParamsNames.INSTRUCTOR_PERMISSION_SUBMIT_SESSION_IN_SECTIONS);
+    }
+
+    @Override
+    public ActionResult execute() {
+        String feedbackResponseId = getNonNullRequestParamValue(Const.ParamsNames.FEEDBACK_RESPONSE_ID);
+
+        FeedbackResponseAttributes response = logic.getFeedbackResponse(feedbackResponseId);
+        Assumption.assertNotNull(response);
+
+        FeedbackResponseCommentSaveRequest comment = getAndValidateRequestBody(FeedbackResponseCommentSaveRequest.class);
```

Comment:
Shouldn't this be `FeedbackResponseCommentCreateRequest`?

Reply:
It's used for both creation and update.

Reply:
Shouldn't it be more clear to have `createRequest` and `updateRequest` separately?